src/crypto/algorithms/megolm.js

```diff
@@ -663,6 +663,89 @@ MegolmDecryption.prototype.onRoomKeyEvent = function(event) {
 };
 
 
+MegolmDecryption.prototype.hasKeysForKeyRequest = function(keyRequest) {
+    const body = keyRequest.requestBody;
+
+    return this._olmDevice.hasInboundSessionKeys(
+        body.room_id,
+        body.sender_key,
+        body.session_id,
+        // TODO: ratchet index
+    );
+};
+
+
+MegolmDecryption.prototype.shareKeysWithDevice = function(keyRequest) {
```

Comment:
They override the methods in the base class - see https://github.com/matrix-org/matrix-js-sdk/blob/2c54d760857116711d2780f9e787533aa6d616d2/src/crypto/algorithms/base.js#L156. I've updated the jsdoc stuff so that the jsdoc will be correctly inherited from the base class.